src/cli/cluster/configure_base_path_proxy.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Server } from 'hapi';
+import { createBasePathProxy } from '../../core';
+import { setupLogging } from '../../server/logging';
+
+export async function configureBasePathProxy(config) {
+  // New platform forwards all logs to the legacy platform so we need HapiJS server
+  // here just for logging purposes and nothing else.
+  const server = new Server();
+  setupLogging(server, config);
+
+  const basePathProxy = createBasePathProxy({ server, config });
+
+  await basePathProxy.configure({
```

Comment:
design question: Why are the args in `configure()` separate from the options passed to `createBasePathProxy()`? Doesn't it just add complexity to the `BasePathProxyRoot` which might not be fully configured when it's methods are called?